src/providers/wms/qgswmsprovider.cpp

```diff
@@ -4455,10 +4466,14 @@ void QgsWmsTiledImageDownloadHandler::tileReplyFinished()
       {
         QgsWmsStatistics::Stat &stat = QgsWmsStatistics::statForUri( mProviderUri );
         stat.errors++;
-
         // if we reached timeout, let's try again (e.g. in case of slow connection or slow server)
-        if ( reply->error() == QNetworkReply::TimeoutError )
-          repeatTileRequest( reply->request() );
+        repeatTileRequest( reply->request() );
+
+        if ( reply->error() == QNetworkReply::ContentAccessDenied )
+        {
+          mError = tr( "Access denied for following reason: %2" ).
```

Comment:
I would shorten this to just "Access denied: %1" (without "for the following reason")

And we should use `%1` instead of `%2` there, right?